app/generate_protos.sh

```diff
@@ -24,6 +24,10 @@ find proto -name '*.proto' | protoc -I proto \
   \
   $(xargs)
 
+# protoc only allows passing --descriptor_set_out once...
+cp proto/protos.pb proxy/protos.pb
+cp proto/protos.pb backend/src/protos.pb
+
```

Comment:
We should only have a single copy of this file and let each one dig it up using lots of ../../..

Does the proxy even use the file?

Reply:
> We should only have a single copy of this file and let each one dig it up using lots of ../../..

They go in different containers; and the container build system won't fetch anything from parent folders. I think it's fine. (Another option might be to symlink, but it's not much better in my opinion).

> Does the proxy even use the file?

Yes, it's required for the GIS stuff to return non-protobuf stuff (i.e. the map layers that are GeoJSON).

Reply:
Symlinks won't work if a subtree is mounted inside the container. The last cp could be an mv so we get one less. And I think backend/src/proto/descriptors.pb is a more descriptive name.

Reply:
Maybe I'm blind, but I don't see proxy/protos.pb referenced anywhere?

Reply:
> Symlinks won't work if a subtree is mounted inside the container.

Docker is stupid. I don't think it is mounted. From memory I think symlinks work! But I'm not 100% sure.


> I think backend/src/proto/descriptors.pb is a more descriptive name.

I agree, will change it.

Reply:
> Maybe I'm blind, but I don't see proxy/protos.pb referenced anywhere? [EDIT] Aha, it was there even before this PR.

Yep, it's in `//app/proxy/envoy.yaml`

Reply:
Btw, I imagine you know this already, but that `protos.pb` (soon to be `descriptors.pb`) is just a serialized `google.protobuf.FileDescriptorSet` message from here `https://github.com/protocolbuffers/protobuf/blob/master/src/google/protobuf/descriptor.proto`.

Reply:
I'd like it to end up inside src/proto/ to keep everything proto-related there, and thus it falls under the generic .gitignore as well.

Reply:
Yes, that's a good idea. I didn't notice, sorry. Want me to fix that or do you have the mutex on this branch now?

Reply:
Please fix it :) Nah, we don't need a mutex, should be easy to rebase my stuff even if you force-push, I just need to learn the comands :D

Reply:
Fixed (I hope)!